src/app/components/champion/ChampionSelector.tsx

```diff
@@ -1,46 +1,57 @@
-import React, {useCallback, useEffect, useState} from "react";
-import {Champion, Data} from "../model/models";
-import {ChampionView} from "./ChampionView";
-import {getUrlChampion, getUrlChampionList} from "../../config/config";
+import React, {useState} from "react";
 import axios from 'axios';
-import {defaultChampion} from "./default";
-
+import {Champion} from "../model/models";
+import {ChampionView} from "./ChampionView";
+import {getUrlChampion, getUrlChampionAvatar, getUrlChampionList} from "../../config/config";
+import {useFetchAPI} from "../../api/reducer";
+import {Autocomplete} from "@material-ui/lab";
+import {Avatar, List, TextField} from "@material-ui/core";
+import {makeStyles} from "@material-ui/core/styles";
+import {extractChampion, extractChampionList} from "../../misc/utils";
 
 export const ChampionSelector = () => {
-    const [champion, setChampion] = useState<Champion>(defaultChampion);
-    const [championsList, setChampionsList] = useState<Champion[]>([]);
+    const [championId, setChampionId] = useState("Aatrox")
+    const championListState = useFetchAPI<Champion[]>([], () => axios.get(getUrlChampionList()), [], extractChampionList)
+    const championState = useFetchAPI<Champion>(undefined, () => axios.get(getUrlChampion(championId)), [championId], extractChampion)
 
-    useEffect(() => {
-        fetchChampionList()
-    }, [])
+    const useStyles = makeStyles(() => ({
+        avatarSearch: {
+            height: '12%',
+            width: '12%',
+            marginRight: '5%'
+        }
+    }));
 
-    function fetchChampionList() {
-        axios.get(getUrlChampionList(), {}).then(response => {
-            const payload = response.data as Data
-            const champs = Object.values(payload.data) as Champion[]
-            setChampionsList(champs)
-        })
-    }
+    const classes = useStyles();
 
-    function loadChampion(championSelected:string) {
-        axios.get(getUrlChampion(championSelected), {}).then(response => {
-            const payload = response.data as Data
-            const champs = Object.values(payload.data) as Champion[]
-            const champion = champs[0]
-            setChampion(champion as Champion)
-        })
+    //has to be an Any for TS compatibility with onChange.
+    function handleTextViewChange(e:any) {
+        if (e) {
```

Comment:
this if is not enough to avoid a an undefined/null error. Also, I dont think that `e.target.firstElementChild.id` is a good idea, probably because handleTextViewChange is not the right function to use on your use case. Watch the documentation carefully pls.

Reply:
Modified the function accordingly (put directly in onInputChange)